core/src/main/java/io/jenkins/pluginhealth/scoring/service/ScoreService.java

```diff
@@ -52,17 +52,31 @@ public Score save(Score score) {
 
     @Transactional(readOnly = true)
     public Optional<Score> latestScoreFor(String pluginName) {
-        return pluginService.findByName(pluginName)
-            .flatMap(repository::findFirstByPluginOrderByComputedAtDesc);
+        return pluginService.findByName(pluginName).flatMap(repository::findFirstByPluginOrderByComputedAtDesc);
+    }
+
+    @Transactional(readOnly = true)
+    public List<Score> getLastTwoScoresSummary() {
+        return repository.findLastTwoScoreForAllPlugins();
     }
 
     @Transactional(readOnly = true)
-    public Map<String, Score> getLatestScoresSummaryMap() {
-        return repository.findLatestScoreForAllPlugins().stream()
-            .collect(Collectors.toMap(
-                score -> score.getPlugin().getName(),
-                score -> score
-            ));
+    public Map<String, Score> getLatestScoresSummaryMap(List<Score> scores) {
+        return scores.stream()
+                .collect(Collectors.toMap(
+                        score -> score.getPlugin().getName(),
+                        score -> score,
+                        (existingScore, newScore) -> existingScore));
+    }
+
+    @Transactional(readOnly = true)
+    public Map<String, Long> getPreviousScoreMap(List<Score> scores) {
+        return scores.stream()
+                .collect(Collectors.toMap(
+                        score -> score.getPlugin().getName(), // Key mapper
+                        Score::getValue, // Value mapper
+                        (existingValue, newValue) -> newValue // Merge function
+                        ));
```

Comment:
I don't understand why the methods have a parameter. 
Those methods don't use the database at all.

Reply:
@alecharp I created this [findLastTwoScoreForAllPlugins](https://github.com/jenkins-infra/plugin-health-scoring/pull/495/files#diff-1ca064d0ec85e9d144d36f8e014e4c106410cfc94c8ee64c5e644e845441a000L56:~:text=List%3CScore%3E%20findLatestScoreForAllPlugins()%3B ) method to fetch last two score for all plugins, and these methods getLatestScoresSummaryMap and getPreviousScoreMap are used to find the last and last second score of a plugin, These methods takes the output generated by indLastTwoScoreForAllPlugins() method. @Transcation annotation need to removed i checked now thanks